package.json

```diff
@@ -117,7 +117,8 @@
     "xterm-addon-fit": "^0.4.0",
     "xterm-addon-search": "^0.7.0",
     "xterm-addon-web-links": "^0.4.0",
-    "xterm-addon-webgl": "^0.9.0"
+    "xterm-addon-webgl": "^0.9.0",
+    "yaml": "^2.0.0"
```

Comment:
is Yaml really required to be added inside the dependencies or is a merge/rebase mistake?

Reply:
Yes, the cloud-init config needs to support YAML comments like the following example. `#cloud-config` needs to parse and save it.

```yaml
#cloud-config
write_files:
  - content: |
      #!/bin/bash
      vmName=$1
      echo "VM Name is: $vmName" > /run/cloud-init/cloudinitscript.log
    path: /run/cloud-init/exec_initscript.sh
    permissions: '0755'
runcmd:
  - - /run/cloud-init/exec_initscript.sh 
    - "{{ ds.meta_data.local-hostname }}"
```

Reply:
Hi， @cnotv   js-yaml doesn't support comment parsing, there are some operations in our UI that need to convert the user input yaml to json.  
eg: If the user checks the checkbox below, we need to know if the user has entered the corresponding value in the above yaml form
![image](https://user-images.githubusercontent.com/24985926/177312787-7d85b35c-e59b-4344-9af9-8bdeffe1f752.png)

However, there are some comments that make sense for cloud-init, but if you convert them via js-yaml, the comments will be lost,  like this  [issue](https://github.com/harvester/harvester/issues/2342#issuecomment-1144809287) 

So we need the yaml package, which duplicates most of the functionality of js-yaml, and to avoid duplicate references, can we consider replacing js-yaml's handling of yaml entirely with the yaml package